### DISCUSSION TITLE
Multiple relations to the same table

### DISCUSSION THREAD

**Discussion by nikolovlazar** [prisma/prisma#3960] [Q&A] [answered]
Hey guys!

I need to build a table that has two references to the same table. The target table is `User`, the source table is `Absence`. The idea is that the Absence table has a reference to the user that's requesting an absence, but I also need to keep a reference to the user who reported it, like their HR representative or similar. The error that happens is: `Automatic related field generation would cause a naming conflict. Please add an explicit opposite relation field.`

Here are my two tables:
```prisma
model User {
  id             Int       @default(autoincrement()) @id
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt
  firstName      String?
  lastName       String?
  email          String    @unique
  role           String    @default("user")
  picture        String?
  sessions       Session[]
}
```

and:

```prisma
model Absence {
  id                 Int       @default(autoincrement()) @id
  createdAt          DateTime  @default(now())
  updatedAt          DateTime  @updatedAt
  approved           Boolean   @default(false)
  type               AbsenceType
  user               User      @relation("user", fields: [userId], references: [id])
  userId             Int
  reporter           User      @relation("reporter", fields: [reporterId], references: [id])<----- ERROR'S HERE
  reporterId         Int
  from               DateTime
  to                 DateTime
  notes              String?
}
```

**Comment by SCD-24** [comment, marked as answer]
```prisma
model ReferCredit {
  id                              String         @id
  referrer                        String?
  referred                        String?        @unique
  User_ReferCredit_referredToUser User?          @relation("ReferCredit_referredToUser", fields: [referred], references: [id])
  User_ReferCredit_referrerToUser User?          @relation("ReferCredit_referrerToUser", fields: [referrer], references: [id])
}

model User {
  id                    String           @id
  ReferredToUser        ReferCredit?     @relation("ReferCredit_referredToUser")
  ReferrerToUser        ReferCredit[]    @relation("ReferCredit_referrerToUser")
}
```

This is what I have set up for something similar, maybe it'll help.  In my case a user can be a referrer and referred.  referrers can have many referred under them. Referred users can only show up once as a referred user.

**Reply by nikolovlazar** [reply]
This does work, but this way I must add the stuff in the User table as well. My idea was to just have a pointer in Absence that points to the user who reported the absence, without having to introduce new properties in the User table. Thanks for the idea. I'll still use it in a different place 😄

**Reply by ryands17** [reply]
@lazarnikolov94 👋 

How about this?

```prisma
model User {
  id        Int       @id @default(autoincrement())
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  firstName String?
  lastName  String?
  email     String    @unique
  role      String    @default("user")
  picture   String?
  reporter  Absence[] @relation("reporter")
  requester Absence[] @relation("user")
}

model Absence {
  id         Int      @id @default(autoincrement())
  createdAt  DateTime @default(now())
  updatedAt  DateTime @updatedAt
  approved   Boolean  @default(false)
  user       User     @relation("user", fields: [userId], references: [id])
  userId     Int
  reporter   User     @relation("reporter", fields: [reporterId], references: [id])
  reporterId Int
  from       DateTime
  to         DateTime
  notes      String?
}

```

This way you can find the reporters and requesters in the User model itself.

**Reply by nikolovlazar** [reply]
Yeah this makes sense. Thank you @ryands17 and @GlennChon!

**Reply by ishubham21** [reply]
How to do this for many-to-many relation? @ryands17 @GlennChon @nikolovlazar

**Reply by elijaholmos** [reply]
Trying to figure out how to do this with many-to-many relations as well

**Reply by serg06** [reply]
@ishubham21 @elijaholmos I posted a many-to-many example below

**Comment by serg06** [comment]
You can do this with many-to-many relationships as well:

```ts
model User {
  id             Int     @id @default(autoincrement())
  likedVideos    Video[] @relation(name: "UserLikedVideos")
  dislikedVideos Video[] @relation(name: "UserDislikedVideos")
}

model Video {
  id        Int    @id @default(autoincrement())
  likers    User[] @relation(name: "UserLikedVideos")
  dislikers User[] @relation(name: "UserDislikedVideos")
}
```

**Reply by orelhochenboym** [reply]
Do you know the syntax for explicit syntax (UsersOnVideos/UsersVideos table)?

**Reply by siddarthkot** [reply]
I am trying to do something like this but prisma doesnt recognize the relationships and throws an error when im trying to connect.

**Reply by janpio** [reply]
This should work as far as I can see. Please open a new discussion if it does not for you @Immortal000 and provide all the information we need to understand.

**Reply by valajczech** [reply]
Thanks, @serg06.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
